lib/Service.js

```diff
@@ -131,8 +131,10 @@ class Service extends Component {
         delete this._pendingRequests[id];
 
         // ** Check for errors
-        if (res.error)
-            return req.fail(res.error);
+        if (res.error){
+            var err = errors.isError(res.error) ? res.error.toObject() : res.error;
```

Comment:
I think this line could throw an error.  The `isError` function is used to detect for Javascript Error objects, not nodus style errors.  With this said, all nodus errors are also Javascript errors.  

So I really like your proposed change, but I think this line prevents errors if the `toObject` method is not defined:
` const err = errors.isError(res.error) ? (util.isFunction(res.error.toObject) ? res.error.toObject() : res.error) : res.error;`